File: SofiForce.Web/ClientApp/src/app/core/Models/SearchModels/SalesOrderSearchModel.ts
import { BranchModel } from "../EntityModels/branchModel";
import { RepresentativeModel } from "../EntityModels/representativeModel";
import { StoreModel } from "../EntityModels/storeModel";
import { BaseSearchModel } from "./baseSearchModel";

export interface SalesOrderSearchModel extends BaseSearchModel {
    salesCode: string;
    clientId:number;
    clientCode: string;
    branchId: number | null;
    branchCode: string | null;
    representativeId: number | null;
    representativeCode:string|null;
    storeId: number | null;
    storeCode: string | null;
    priorityTypeId: number | null;
    salesDate: Date | null;
    salesOrderStatusId: number;
    salesOrderTypeId:number;
    salesOrderSourceId:number [];
    invoiceDate: Date | null;
    isInvoiced:number;
    filterMode:number | 1;
    cashDiscountOnly:boolean |0;
}

File: SofiForce.Models/Models/SearchModels/SalesOrderSearchModel.cs
using System;
using System.Collections.Generic;

namespace Models
{
    public class SalesOrderSearchModel : BaseSearchModel
    {
		public Int32? ClientId { get; set; }
		public string SalesCode { get; set; }
		public string ClientCode { get; set; }

		public Int32? BranchId { get; set; }
		public Int32? RepresentativeId { get; set; }

		public Int32? StoreId { get; set; }

		public Int32? PriorityTypeId { get; set; }

		public DateTime? SalesDate { get; set; }
		public Int32? SalesOrderStatusId { get; set; }
		public List<int>? SalesOrderSourceId { get; set; }
		public Int32? SalesOrderTypeId { get; set; }
		public DateTime? InvoiceDate { get; set; }

		public Int32? IsInvoiced { get; set; }

		public int FilterMode { get; set; }

		public bool? CashDiscountOnly { get; set; }=false;

	}
}